packages/process/process_monitor.pony

```diff
@@ -107,6 +108,7 @@ type ProcessError is
   | WaitpidError
   | WriteError
   | CapError
+  | (PreExecError, U8, I32)
```

Comment:
i would prefer the previous solution of two different error primitives for chdir and exec.
The best case would be if we could make these be two primitives that do not contain the, that would be most easy for matching on the result. Having a tuple here seems to be a little bit odd.

So my suggestions for representing execve- and chdir-errors in descending appreciation:

1. reuse existing `ExecveError` and add `primitive ChDirError` (though we lose the errorcode information, but keep "compatibility" with the existing `ProcessError`)
2. `(primitive ExecveError, I32)` and `(primitive ChdirError, I32)`
3. `class ExecveError(error: I32)` and `class ChdirError(error: I32)`

What do you think @cquinn @jemc @rkallos ?

Reply:
I have no strong opinions one way or the other on this one.

Reply:
I'm a little reluctant to reuse `ExecveError` so much, but I don't really know if my feelings are reasonable in this case.

I think we'll all agree that path of least resistance is to reuse `ExecveError' and make a new `ChdirError` primitive. I'll try to find some spare cycles and write it up this weekend.

Thanks again for all your help!